[Model-Monitoring] Fix TimescaleDB get-records

alxtkr77 opened the pull request:
### 📝 Description

  Fixes ML-11516 by ensuring the `_get_records()` API always includes the `endpoint_id` column in results for metrics and app_results tables. Without this column, it was impossible to identify which endpoint each record belongs to.

  Additionally improves the API by using `TimescaleDBTables` enum constants instead of string literals, making the API more consistent and type-safe.

  ---

  ### 🛠️ Changes Made

  **Bug Fix:**
  - Add `endpoint_id` to hardcoded columns list in `read_metrics_data_impl()` (metrics table)
  - Add `endpoint_id` to hardcoded columns list in `read_results_data_impl()` (app_results table)
  - Predictions table already includes `endpoint_id` via table schema (no changes needed)

  **API Improvement:**
  - Replace string literals with `TimescaleDBTables` enum constants (METRICS, APP_RESULTS, PREDICTIONS)
  - Update docstring to reference enum usage
  - Update error message to show enum constant names

  **Test Coverage:**
  - Add `endpoint_id` assertions to all existing `_get_records()` tests
  - Add new test for column filtering with `endpoint_id` explicitly requested
  - Remove redundant `isinstance()` and `empty` checks per testing standards
  - All assertions reference test data (no hardcoded values)

  **Files Modified:**
  - `mlrun/model_monitoring/db/tsdb/timescaledb/queries/timescaledb_metrics_queries.py`
  - `mlrun/model_monitoring/db/tsdb/timescaledb/queries/timescaledb_results_queries.py`
  - `mlrun/model_monitoring/db/tsdb/timescaledb/timescaledb_connector.py`
  - `tests/model_monitoring/db/tsdb/timescaledb/test_timescaledb_get_records.py`

  ---

  ### ✅ Checklist
  - [x] I updated the documentation (docstrings updated)
  - [x] I have tested the changes in this PR

  ---

  **Before Fix:**
  ```python
  df = connector._get_records(table="metrics", ...)
  # Result columns: [timestamp, application_name, metric_name, metric_value]
  # Missing: endpoint_id - can't identify which endpoint!

  After Fix:
  df = connector._get_records(table=TimescaleDBTables.METRICS, ...)
  # Result columns: [timestamp, endpoint_id, application_name, metric_name, metric_value]
  # Now includes endpoint_id - can identify records by endpoint!

  ---
  🔗 References

  - Jira: https://jira.iguazeng.com/browse/ML-11516 - [timescaleDB] Align record output like tdengine when retrieve MEP record
  - Related: PR #8926 (added _get_records() API)

  ---
  🚨 Breaking Changes?

  - No

Commit:
[Model Monitoring] Add comprehensive tests for endpoint_id in _get_records()

- Add assertions to verify endpoint_id is always present in results
- Add test for endpoint_id when explicitly requested in columns parameter
- Remove redundant isinstance/empty checks (per testing standards)
- All assertions reference test data, not hardcoded values
- Tests now would have caught the missing endpoint_id bug

Commit:
[Model Monitoring] Use TimescaleDBTables enum constants in _get_records()

- Replace string literals with enum constants (METRICS, APP_RESULTS, PREDICTIONS)
- Update docstring to reference enum usage
- Update error message to show enum constant names
- Update all tests to use enum constants
- Improves consistency and allows users to use constants directly

assaf758 commented on tests/model_monitoring/db/tsdb/timescaledb/test_timescaledb_get_records.py:
```diff
@@ -264,8 +269,58 @@ def test_get_records_with_column_filter(self, connector, query_test_helper):
+        # Verify other unrequested columns are also NOT included
+        assert mm_schemas.WriterEvent.APPLICATION_NAME not in df.columns
+
+    def test_get_records_with_endpoint_id_in_columns(
```
do we really need a new test for this? why not adjust existing ones?

alxtkr77 replied:
Good point! I've removed the redundant test.

The existing `test_get_records_with_column_filter` already verifies that:
- Column filtering works correctly
- endpoint_id is properly excluded when not in the columns list
- Other columns are also properly filtered

There's no need for a separate test to verify endpoint_id can be explicitly included - that's just standard column filtering behavior already covered by the existing test.

Thanks for the feedback!

Commit:
[Model Monitoring] Add start_infer_time to _get_records() output

Include start_infer_time column in metrics and app_results tables output.

Changes:
- Add START_INFER_TIME to read_metrics_data_impl() columns list
- Add START_INFER_TIME to read_results_data_impl() columns list
- Extend existing tests to verify START_INFER_TIME is present
- Remove redundant test_get_records_with_endpoint_id_in_columns test per PR review feedback

Reference: ML-11516
